### ISSUE TITLE
[Feature] Separate sql parser to a independent module

### ISSUE THREAD

**Issue by zhaomin1423** [apache/doris#9671]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

SQL parser is widely used for permission control and blood relationship analysis, a separate module can be used more conveniently.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by morrySnow** [comment]
Hi, separated sql parser model is not a good idea to do that. A better way to do that is a common framework for getting query plans and making changes to them. Under this framework, we can not only support authentication and lineage, but also other things such as auditor and query statistics.

@EmmyMiao87 PTAL.

**Comment by zhaomin1423** [comment]
yeah, a new framework is a good idea, but, its goal is to have a single package in maven  for users to use.

**Comment by codegit10001** [comment]
啥时候能支持，很期待这个功能
